url6f34cc586e_mekku93_sxl_interpreter_repl_tgz-pJ8-com_miguel_sxl_SXLJ8/src/com/miguel/sxl/SXLParser.java

```diff
@@ -0,0 +1,2739 @@
+                System.out.println("Reading from standard input...");
+                System.out.println();
+                SXLParser parser = new SXLParser(System.in);
+                try {
+                        SimpleNode n = parser.Start();
+                        n.dump("");
+                        System.out.println("Thank you.");
+                }
+                catch (Exception e) {
+                        System.out.println("Oops.");
+                        System.out.println(e.getMessage());
+                }
+        }
+
+/****************************************************************************
+ *    NON-TERMINALS															*
+ ****************************************************************************/
+
+
+
+/**== OPERATORS ==*/
+  final public void AddOp() throws ParseException {
+               /*@bgen(jjtree) AddOp */
+  ASTAddOp jjtn000 = new ASTAddOp(JJTADDOP);
+  boolean jjtc000 = true;
+  jjtree.openNodeScope(jjtn000);
+  jjtn000.jjtSetFirstToken(getToken(1));
+    try {
+      if (jj_2_1(2)) {
+        jj_consume_token(PLUS);
+      } else if (jj_2_2(2)) {
+        jj_consume_token(MINUS);
+      } else if (jj_2_3(2)) {
+        jj_consume_token(KW_OR);
+      } else {
+        jj_consume_token(-1);
+        throw new ParseException();
+      }
+    } finally {
+          if (jjtc000) {
+            jjtree.closeNodeScope(jjtn000, true);
+            jjtn000.jjtSetLastToken(getToken(0));
+          }
+    }
+  }
+
+  final public void MultOp() throws ParseException {
+                /*@bgen(jjtree) MultOp */
+  ASTMultOp jjtn000 = new ASTMultOp(JJTMULTOP);
+  boolean jjtc000 = true;
+  jjtree.openNodeScope(jjtn000);
+  jjtn000.jjtSetFirstToken(getToken(1));
+    try {
+      if (jj_2_4(2)) {
+        jj_consume_token(MULTIPLY);
+      } else if (jj_2_5(2)) {
+        jj_consume_token(DIVIDE);
+      } else if (jj_2_6(2)) {
+        jj_consume_token(KW_AND);
+      } else {
+        jj_consume_token(-1);
+        throw new ParseException();
+      }
+    } finally {
+          if (jjtc000) {
+            jjtree.closeNodeScope(jjtn000, true);
+            jjtn000.jjtSetLastToken(getToken(0));
+          }
+    }
+  }
+
+  final public void RelOp() throws ParseException {
+               /*@bgen(jjtree) RelOp */
+  ASTRelOp jjtn000 = new ASTRelOp(JJTRELOP);
+  boolean jjtc000 = true;
+  jjtree.openNodeScope(jjtn000);
+  jjtn000.jjtSetFirstToken(getToken(1));
+    try {
+      if (jj_2_7(2)) {
+          ASTGreaterThan jjtn001 = new ASTGreaterThan(JJTGREATERTHAN);
+          boolean jjtc001 = true;
+          jjtree.openNodeScope(jjtn001);
+          jjtn001.jjtSetFirstToken(getToken(1));
+        try {
+          jj_consume_token(GREATER);
+        } finally {
+          if (jjtc001) {
+            jjtree.closeNodeScope(jjtn001, true);
+            jjtn001.jjtSetLastToken(getToken(0));
+          }
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 89:96, 102:109, 115:122, 128:135, 141:148, 154:161, 180:187, 196:203, 209:216, 222:229, 235:242, 248:255, 261:268, 288:295, 301:308, 314:321, 327:334, 340:347, 353:360, 380:387.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.